src/elysium/signaturebuilder.cpp

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2020 The Zcoin Core Developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include "convert.h"
+#include "sigmaprimitives.h"
+#include "signaturebuilder.h"
+
+#include "../base58.h"
+
+#include <secp256k1/include/Scalar.h>
+
+namespace elysium {
+
+SigmaV1SignatureBuilder::SigmaV1SignatureBuilder(
+    CBitcoinAddress const &receiver,
+    int64_t referenceAmount,
+    SigmaProof const &proof,
+    CPubKey const &publicKey)
+    : hasher(CHashWriter(SER_GETHASH, PROTOCOL_VERSION)), publicKey(publicKey)
+{
+    // serialize payload
+    CKeyID keyId;
+    if (!receiver.GetKeyID(keyId)) {
+        throw std::runtime_error("Fail to get address key id.");
+    }
+
+    hasher.write(reinterpret_cast<char*>(keyId.begin()), keyId.size());
+
+    // reference amount
+    elysium::swapByteOrder(referenceAmount);
```

Comment:
Why do we need to switch endianness?

Reply:
The endianness would be the same with datas that we store to payload.